src/components/Map/Layers/Buildings/HelsinkiBuildings/Popup.tsx

```diff
@@ -2,6 +2,51 @@ import React from 'react'
 import Feature from 'ol/Feature'
 import { Table, TableBody, TableCell, TableRow } from '@mui/material'
 import _ from 'lodash'
+// Variables
+var heatings
+var tecons
+var teconss
+var emisfactord 
+var emisfactords 
+// convert to float
+function convert_to_float(a) {
+  // of string to float
+  var floatValue = +(a);   
+  // Return float value
+  return floatValue;
+}
+// Emission factors [kgCO2/kWh]
+var empdp = [0.255,0.195,0.104,0.104,0.104]
+// empdp[0]
+// round the tecons value
+function Tecons(tecc) {
+  tecons = Math.round(tecc/1000)*1000
+  teconss = tecons.toString().slice(0, -3);
+  teconss = `${teconss} 000`
+  return teconss
+}
+// Emission factor function [kgCO2/kWh]
+function emissionFactor(tecc, empdpn) {  
+  emisfactord = Math.round((tecc * empdpn)/1000)*1000    
+  emisfactords = emisfactord.toString().slice(0, -3);  
+  emisfactords = `${emisfactords} 000`
+  return emisfactords;
+}
+// Estimated annual energy consumption [kWh/m3,a]
```

Comment:
It seems a bit redundant to transform the JSON-string to javascript object every time the code is run. Could you pre-convert the json to object? You can use the console on any webpage to run this code and just paste the resulting object here.

Also, please split large constant objects to a "constants.ts"-file and export from there. See [https://github.com/AvoinOrg/climate-map/blob/ol/src/components/Map/Layers/Forests/FinlandMatureForests/constants.ts](https://github.com/AvoinOrg/climate-map/blob/ol/src/components/Map/Layers/Forests/FinlandMatureForests/constants.ts)